src/main.rs

```diff
@@ -0,0 +1,59 @@
+use std::env;
+
+use error::VMError;
+use hardware::{CondFlag, Memory, OpCode, Register, Registers};
+use instructions::*;
+use utils::{load_arguments, setup, shutdown};
+
+mod error;
+mod hardware;
+mod instructions;
+mod trap_routines;
+mod utils;
+
+const PC_START: u16 = 0x3000;
+
+fn main() -> Result<(), VMError> {
+    let mut args = env::args();
+    // Hardware creation
+    let mut regs = Registers::new();
+    let mut mem = Memory::new();
+    // Read the file with the instructions to execute into the VM's memory
+    load_arguments(&mut args, &mut mem)?;
+    // Setup of:
+    // - Terminal
+    // - Register Cond: It is set to the Zro flag so as to set a value
+    // - Program Counter: It is set to the PC_START which is a default address
+    let termios = setup()?;
+    regs[Register::Cond] = CondFlag::Zro.value();
+    regs[Register::PC] = PC_START;
+
+    // VM main loop
+    let mut running_flag = true;
+    while running_flag {
```

Comment:
Its a good idea to not have domain specific logic in the `main` function. Since it cannot be reused.

The fetch-decode-execute cycle should not be here (it belongs to the library code), that will allow you to reuse that logic in different places (integration tests, for example).

The main function should only manage argument parsing and delegate to library functions.